aten/src/ATen/native/native_functions.yaml

```diff
@@ -9289,3 +9289,8 @@
 - func: _test_ambiguous_defaults.b(Tensor dummy, int a=2, str b="2") -> Tensor
   cpp_no_default_args: ['a', 'b']
   python_module: nn
+
+- func: segment_reduce(Tensor data, str reduce, Tensor? lengths=None, Tensor? indices=None, int axis=0, bool unsafe=False) -> Tensor
```

Comment:
you probably want all the args except `data` to be kwarg-only, so do it like
```
segment_reduce(Tensor data, *, str reduce ....)
```

Reply:
Is there an example I can follow for this? Also, where would the default be set in this approach?